lib/src/main/java/co/nimblehq/alpine/lib/mrz/MrzManager.kt

```diff
@@ -0,0 +1,145 @@
+@file:JvmName("MrzManager")
+
+package co.nimblehq.alpine.lib.mrz
+
+import android.graphics.BitmapFactory
+import android.util.Log
+import co.nimblehq.alpine.lib.mrz.MrzProcessorException.*
+import com.google.mlkit.common.MlKitException
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.*
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import java.io.File
+import java.util.regex.Pattern
+
+interface ResultListener {
+    fun onSuccess(mrzInfo: MrzInfo)
+
+    fun onError(e: MrzProcessorException)
+}
+
+sealed class MrzProcessorException : Exception() {
+    object DefaultMrzProcessorException : MrzProcessorException()
+    object InvalidMrzInfoMrzProcessorException : MrzProcessorException()
+    object MrzInfoNotFoundMrzProcessorException : MrzProcessorException()
+    object FileNotFoundMrzProcessorException : MrzProcessorException()
+    object TextNotFoundMrzProcessorException : MrzProcessorException()
+    object TextNotRecognizedMrzProcessorException : MrzProcessorException()
+}
+
+interface MrzProcessor {
+    fun processImageFile(filePath: String, resultListener: ResultListener)
+
+    companion object {
+        @JvmStatic
+        fun newInstance(): MrzProcessor = MrzProcessorImpl()
+    }
+}
+
+private const val TAG = "MlKitWrapper"
+private const val PASSPORT_TD_3_LINE_1_REGEX = "(P[A-Z0-9<]{1})([A-Z]{3})([A-Z0-9<]{39})"
+private const val PASSPORT_TD_3_LINE_2_REGEX =
+    "([A-Z0-9<]{9})([0-9]{1})([A-Z]{3})([0-9]{6})([0-9]{1})" +
+        "([M|F|X|<]{1})([0-9]{6})([0-9]{1})([A-Z0-9<]{14})([0-9]{1})([0-9]{1})"
+private const val IMAGE_ORIENTATION_IN_DEGREES = 90
+private const val DOCUMENT_NUMBER_MINIMUM_LENGTH = 8
+private const val DATE_OF_BIRTH_MINIMUM_LENGTH = 6
+private const val DATE_OF_EXPIRY_MINIMUM_LENGTH = 6
+
+@Suppress("NestedBlockDepth")
+private class MrzProcessorImpl : MrzProcessor {
```

Comment:
Should we rename `MrzProcessorImpl` to `MrzManagerImpl`, so the `MrzProcessor` should be `MrzManager` to make it consistent with the file name.
```suggestion
private class MrzManagerImpl : MrzManager {
```

Reply:
@Wadeewee it'll give build error - because of `@file:JvmName("NfcManager")`.  Without `@file:JvmName("NfcManager")`, apps using Java classes will see `NfcManagerKt` file, which is not the best thing to see as a consumer of the library. In fact, this is the very reason I changed the file name to `NfcManager`. What do you think?

Reply:
@AVI5HEK Why would java classes see `MrzManagerKt` if there are no public file level properties or methods to access? 🤔 Seems the only public things in there are interfaces which shouldn't show an `xxxKt` file 🤔

Reply:
> it'll give build error - because of @file:JvmName("MrzManager"). 

@AVI5HEK I tried to comment `@file:JvmName("MrzManager")`, but I didn't get any error. I still can build and run the app 🤔 Not sure we need to use it.

I think it makes sense to naming as `MrzManager` and `NfcManager` as we will call them in another module, So we should call or interact with the `Manager` class 🤔

Reply:
@ryan-conway any idea how to avoid this?
![image](https://user-images.githubusercontent.com/8093908/179697837-ce43052e-4557-4fd2-b2e7-ce27e9be33ae.png)